### PR TITLE
Nc feat/one to one

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

## Summary by CodeRabbit

- **New Features**
	- Introduced a Vue component for managing one-to-one relationships, enhancing user interaction with related table metadata and display options.
	- Added support for creating unique constraints in MySQL, PostgreSQL, and SQLite queries, improving data integrity and uniqueness enforcement.
	- Introduced new methods and logic to handle one-to-one relationships across various components and services, including the ability to count excluded one-to-one children records and manage 'oo' type relations.
	- Added a new endpoint for excluding certain data based on parameters, enhancing data retrieval flexibility.
- **Enhancements**
	- Refined logic related to column types and offset calculations in multiple components for better clarity and usability.
	- Enhanced handling of relation types in query building and condition parsing, ensuring correct processing paths based on relation types.
	- Improved handling and deletion of 'oo' relations within services, contributing to more robust data management.
- **Bug Fixes**
	- Adjusted conditions for wrapping values in arrays for lookup columns and handling one-to-one relationships, fixing issues with incorrect data formatting and display.
- **Documentation**
	- Added new examples in API documentation to illustrate usage and improve developer understanding.

### CODE REVIEW

File: packages/nocodb/src/services/columns.service.ts [hunk 1]
```diff
@@ -2623,6 +2782,103 @@ export class ColumnsService {
...
+        parent,
+        childColumn,
+        (param.column as LinkToAnotherColumnReqType).type as RelationTypes,
+        (param.column as LinkToAnotherColumnReqType).title,
+        foreignKeyName,
+        (param.column as LinkToAnotherColumnReqType).virtual,
+        null,
+        param.column['meta'],
+        param.colExtra,
+      );
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/services/columns.service.ts:2881]: It's already validated at the beginning

File: packages/nocodb/src/db/sql-client/lib/mysql/MysqlClient.ts [hunk 2]
```diff
@@ -2496,6 +2496,7 @@ class MysqlClient extends KnexClient {
     query += n.un ? ' UNSIGNED' : '';
     query += n.rqd ? ' NOT NULL' : ' NULL';
     query += n.ai ? ' auto_increment' : '';
+    query += n.unique ? ` UNIQUE` : '';
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/db/sql-client/lib/mysql/MysqlClient.ts:2499]: sanitisation is done using knex wherever it's necessary.

File: packages/nocodb/src/db/sql-client/lib/pg/PgClient.ts [hunk 3]
```diff
@@ -2863,6 +2863,7 @@ class PGClient extends KnexClient {
         );
         query += n.rqd ? ' NOT NULL' : ' NULL';
         query += defaultValue ? ` DEFAULT ${defaultValue}` : '';
+        query += n.unique ? ` UNIQUE` : '';
       }
     } else if (change === 1) {
       query += this.genQuery(
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/db/sql-client/lib/pg/PgClient.ts:2869]: sanitisation is done using knex wherever it's necessary.

File: packages/nocodb/src/services/data-alias-nested.service.ts [hunk 4]
```diff
@@ -186,6 +186,47 @@ export class DataAliasNestedService {
...
+        cid: param.rowId,
+      },
+      param.query,
+    );
+
+    return new PagedResponseImpl(data, {
+      count,
+      ...param.query,
+    });
+  }
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/services/data-alias-nested.service.ts:229]: Any unhandled errors are captured globally using Nestjs filter.

File: packages/nocodb/src/helpers/columnHelpers.ts [hunk 5]
```diff
@@ -91,6 +91,73 @@ export async function createHmAndBtColumn(
...
+      fk_parent_column_id: parent.primaryKey.id,
+      fk_related_model_id: child.id,
+      virtual,
+      system: isSystemCol,
+      fk_col_name: fkColName,
+      fk_index_name: fkColName,
+      meta,
+      ...(colExtra || {}),
+    });
+  }
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/helpers/columnHelpers.ts:159]: `Error Handling` - is done globally in nestjs global exception filter.

`Code Duplication` - planning to refactor in a followup PR.

File: packages/nocodb/src/services/columns.service.ts [hunk 6]
```diff
@@ -31,6 +31,7 @@
 import ProjectMgrv2 from '~/db/sql-mgr/v2/ProjectMgrv2';
 import {
   createHmAndBtColumn,
+  createOOColumn,
   generateFkName,
   randomID,
   sanitizeColumnName,
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/services/columns.service.ts:37]: Error handling is done globally most of the cases

File: packages/nocodb/src/db/sql-client/lib/pg/PgClient.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/db/sql-client/lib/pg/PgClient.ts:2869]: We are using `genQuery` method from base class for sanitisation of user input.

**Review by dstala — APPROVED** [review]

**Pull request merged**